api/utils/keys/yubikey.go

```diff
@@ -144,6 +165,9 @@ func (y *YubiKeyPrivateKey) Public() crypto.PublicKey {
 
 // Sign implements crypto.Signer.
 func (y *YubiKeyPrivateKey) Sign(rand io.Reader, digest []byte, opts crypto.SignerOpts) ([]byte, error) {
+	signCtx, cancel := context.WithCancel(context.Background())
```

Comment:
Similarly, could we add `ctx` to the method signature instead?

Reply:
If we did, it wouldn't implement `crypto.Signer`. We would need to store a context in the struct which isn't really what we want either. I haven't found any solution like this that I'm happy with though.

Reply:
I would take a comment here too, but otherwise it's alright.